libsolutil/CommonData.h

```diff
@@ -295,6 +310,43 @@ decltype(auto) mapTuple(Callable&& _callable)
 	return detail::MapTuple<Callable>{std::forward<Callable>(_callable)};
 }
 
+namespace detail
+{
+
+template<typename Container, typename Value>
+auto findOffset(Container&& _container, Value&& _value, int)
+-> decltype(_container.find(_value) == _container.end(), std::optional<size_t>())
+{
+	auto it = _container.find(std::forward<Value>(_value));
+	auto end = _container.end();
+	if (it == end)
+		return std::nullopt;
+	return std::distance(it, end);
```

Comment:
Why from it to end?

Reply:
Ah, good catch! I guess I had it work backwards in both cases at some point and this version is actually not used anymore currently.